librocksdb_sys/crocksdb/c.cc

```diff
@@ -543,13 +543,16 @@ static bool SaveError(char** errptr, const Status& s) {
   assert(errptr != nullptr);
   if (s.ok()) {
     return false;
-  } else if (*errptr == nullptr) {
-    *errptr = strdup(s.ToString().c_str());
-  } else {
+  }
+
+  if (*errptr != NULL && (long)*errptr != 0x1L) {
     // TODO(sanjay): Merge with existing error?
     // This is a bug if *errptr is not created by malloc()
     free(*errptr);
-    *errptr = strdup(s.ToString().c_str());
+  }
+  *errptr = strdup(s.ToString().c_str());
+  if (*errptr == NULL) {
+    *errptr = (char*)0x1L;
```

Comment:
make a global const empty string, and point errptr there instead of 0x1?

Reply:
It's also a choice. The key is print a log for this path. We can print the log here directly instead of print in caller.

The bad news is that the PR can't fix a problem as we expected. So I'm going to close it.